分区 基础知识 的第 166 页

示例
Genome assembly report (nih.gov)数据结构描述

命令行运行：datasets summary genome accession XXXX

1.

GCA_002947415.1（原始数据第1行）
(ncbi_datasets) [liuchangle@localhost ~]$ datasets summary genome accession GCA_002947415.1
{"reports":

[{"accession":"GCA_002947415.1",
    "annotation_info":

{"method":"Best-placed reference protein set; GeneMarkS+",
"name":"NCBI Prokaryotic Genome Annotation Pipeline (PGAP)",
"pipeline":"NCBI Prokaryotic Genome Annotation Pipeline (PGAP)",
"provider":"NCBI",
"release_date":"2017-12-07",
"software_version":"4.3",
"stats":{

"gene_counts":

{"non_coding":94,转录的非编码基因（例如 lncRNA、miRNA、rRNA 等）的计数；不包括转录的假基因

"protein_coding":3925,已注释结构基因有3925个

"pseudogene":133,转录和非转录假基因的计数133个

"total":4152} 已注释基因有4152个

}

},
"assembly_info":

{"assembly_level":"Complete Genome",基因组组装的水平；

基因组组装一般分为三个层次，contig, scaffold和chromosomes.
"assembly_method":"canu v. 1.6",
"assembly_name":"ASM294741v1",
"assembly_status":"current",

"assembly_type":"haploid",单倍体
"bioproject_accession":"PRJNA292904",
"bioproject_lineage":
[{"bioprojects":

[{"accession":"PRJNA292904",
"parent_accessions":["PRJNA294416"],
"title":"Gram Negative Carbapenemase Detection Genome sequencing and assembly"
},
{"accession":"PRJNA294416",
"parent_accessions":["PRJNA965738"],
"title":"FDA-CDC Antimicrobial Resistance Bank"
},
{"accession":"PRJNA965738",
"title":"CDC DHQP Seq"
}]

}],
"biosample":

{"accession":"SAMN04014897",
"attributes":

[{"name":"isolation_source","value":"missing"},
{"name":"lat_lon","value":"missing"},
{"name":"collected_by","value":"missing"},
{"name":"host_disease","value":"missing"},
{"name":"host","value":"missing"},
{"name":"isolate","value":"missing"},
{"name":"geo_loc_name","value":"missing"},
{"name":"collection_date","value":"missing"},
{"name":"panel_id","value":"CarbaNP-25"},
{"name":"strain","value":"AR_0056"}
],
"bioprojects":

[{"accession":"PRJNA292904"}],

"description":

{"organism":

{"organism_name":"Acinetobacter baumannii","tax_id":470},

分区 基础知识 的第 167 页

{"organism_name":"Acinetobacter baumannii","tax_id":470},

"title":"Pathogen: clinical or host-associated sample from Acinetobacter baumannii"
},

"last_updated":"2018-02-12T10:25:19.737",
"models":["Pathogen.cl"],
"owner":

{"contacts":[{}],
"name":"Centers for Disease Control and Prevention"},

"package":"Pathogen.cl.1.0",
"publication_date":"2015-08-27T00:00:00.000",
"sample_ids":

[{"label":"Sample name","value":"FDA-CDC-AR_0056"},{"db":"SRA","value":"SRS1350019"}],

"status":{"status":"live","when":"2016-12-21T12:18:21"},
"submission_date":"2015-08-27T16:16:14.000"
},
"comments":

"Annotation was added by the NCBI Prokaryotic Genome Annotation Pipeline (released 2013). Information about the Pipeline can
be found here: https://www.ncbi.nlm.nih.gov/genome/annotation_prok/","paired_assembly":{"accession":"GCF_
002947415.1","annotation_name":"NCBI Prokaryotic Genome Annotation Pipeline (PGAP)",

"status":"current"
},

"release_date":"2018-02-18",
"sequencing_tech":"PacBio RSII",
"submitter":"Centers for Disease Control and Prevention"
},

"assembly_stats":

{"contig_l50":1,"contig_n50":3993843,

"gc_count":"1638121","gc_percent":39, CG含量

"genome_coverage":"150.0x",基因组组装覆盖率
"number_of_component_sequences":4,
"number_of_contigs":4,"number_of_scaffolds":4,
"scaffold_l50":1,"scaffold_n50":3993843,

"total_number_of_chromosomes":4,提交到基因组组装中的核染色体、细胞器和质粒计数

"total_sequence_length":"4188385",核基因组的总序列长度，包括未定位和未定位的序列
"total_ungapped_length":"4188385"},

"average_nucleotide_identity":
{"best_ani_match":

{"ani":99.63,"assembly":"GCA_018123985.1","assembly_coverage":89.27,"category":"type","organism_name":"Acinetobacter
baumannii","type_assembly_coverage":93.93},

"category":"category_na",
"comment":"na",
"match_status":"species_match",
"submitted_ani_match":

{"ani":99.63,"assembly":"GCA_018123985.1","assembly_coverage":89.27,"category":"type","organism_name":"Acinetobacter
baumannii","type_assembly_coverage":93.93},

"submitted_organism":"Acinetobacter baumannii",
"submitted_species":"Acinetobacter baumannii",
"taxonomy_check_status":"OK"
},
"checkm_info":

{"checkm_marker_set":"Acinetobacter baumannii",
"checkm_marker_set_rank":"species",
"checkm_species_tax_id":470,
"checkm_version":"v1.2.2",
"completeness":99.15,
"completeness_percentile":31.935078,
"contamination":0.17
},

"current_accession":"GCA_002947415.1",
"organism":

{"infraspecific_names":

{"strain":"AR_0056"},

"organism_name":"Acinetobacter baumannii",
"tax_id":470
},

"paired_accession":"GCF_002947415.1",
"source_database":"SOURCE_DATABASE_GENBANK"
}],

"total_count": 1}

2.

GCF_001701355.1 （原始数据第1228行）
{"reports": [{"accession":"GCF_001701355.1","annotation_info":{"method":"Best-placed reference protein set; GeneMarkS-2+","name":"NCBI
Prokaryotic Genome Annotation Pipeline (PGAP)","pipeline":"NCBI Prokaryotic Genome Annotation Pipeline (PGAP)","provider":"NCBI
RefSeq","release_date":"2023-08-09","software_version":"6.5","stats":{"gene_counts":{"non_coding":121,"protein_coding":5435,"pseudogene":132,"t
otal":5688}}},"assembly_info":{"assembly_level":"Contig","assembly_method":"SPAdes v.
3.6","assembly_name":"ASM170135v1","assembly_status":"current","assembly_type":"haploid","bioproject_accession":"PRJNA307517","bioproject_line
age":[{"bioprojects":[{"accession":"PRJNA307517","title":"Multiple species Genome sequencing and
assembly"}]}],"biosample":{"accession":"SAMN04868739","attributes":[{"name":"lat_lon","value":"38.03  N 23.79
E"},{"name":"collection_date","value":"2014-06-05"},{"name":"collected_by","value":"Hygeia General
Hospital"},{"name":"genotype","value":"ST258"},{"name":"host_disease","value":"missing"},{"name":"isolation_source","value":"stool"},{"name":"strain",
"value":"19_GR_14"},{"name":"host","value":"Homo  sapiens"},{"name":"geo_loc_name","value":"Greece:
Athens"}],"bioprojects":[{"accession":"PRJNA307517"}],"description":{"organism":{"organism_name":"Klebsiella
pneumoniae","tax_id":573},"title":"Pathogen: clinical or host-associated sample from Klebsiella
pneumoniae"},"last_updated":"2021-02-24T10:14:37.074","models":["Pathogen.cl"],"owner":{"contacts":[{}],"name":"The  University of
Queensland"},"package":"Pathogen.cl.1.0","publication_date":"2016-04-19T00:00:00.000","sample_ids":[{"label":"Sample name","value":"19_GR_
14"},{"db":"SRA","value":"SRS4391562"}],"status":{"status":"live","when":"2016-08-09T13:07:16"},"submission_date":"2016-04-19T00:18:07.000"},"com
ments":"Annotation was added by the NCBI Prokaryotic Genome Annotation Pipeline (released 2013). Information about the Pipeline can be found here:
https://www.ncbi.nlm.nih.gov/genome/annotation_prok/","paired_assembly":{"accession":"GCA_001701355.1","annotation_name":"NCBI Prokaryotic
Genome Annotation Pipeline (PGAP)","status":"current"},"release_date":"2016-08-09","sequencing_tech":"Illumina MiSeq","submitter":"The Unniversity
of

分区 基础知识 的第 168 页

of
Queensland"},"assembly_stats":{"contig_l50":7,"contig_n50":299505,"gc_count":"3212347","gc_percent":57,"genome_coverage":"100.0x","number_
of_component_sequences":89,"number_of_contigs":89,"total_sequence_length":"5623115","total_ungapped_length":"5623115"},"average_nucleotid
e_identity":{"best_ani_match":{"ani":99.95,"assembly":"GCA_000788015.1","assembly_coverage":97.99,"category":"type","organism_name":"Klebsiella
pneumoniae","type_assembly_coverage":96.23},"category":"category_na","comment":"na","match_status":"derived_species_match","submitted_ani_m
atch":{"ani":99.95,"assembly":"GCA_000788015.1","assembly_coverage":97.99,"category":"type","organism_name":"Klebsiella pneumoniae subsp.
pneumoniae","type_assembly_coverage":96.23},"submitted_organism":"Klebsiella pneumoniae","submitted_species":"Klebsiella
pneumoniae","taxonomy_check_status":"OK"},"checkm_info":{"checkm_marker_set":"Klebsiella
pneumoniae","checkm_marker_set_rank":"species","checkm_species_tax_id":573,"checkm_version":"v1.2.2","completeness":98.78,"completeness_perc
entile":66.96832,"contamination":0.28},"current_accession":"GCF_001701355.1","organism":{"infraspecific_names":{"strain":"19_GR_
14"},"organism_name":"Klebsiella pneumoniae","tax_id":573},"paired_accession":"GCA_
001701355.1","source_database":"SOURCE_DATABASE_REFSEQ","wgs_info":{"master_wgs_url":"https://www.ncbi.nlm.nih.gov/nuccore/LXMK00000000.
1","wgs_contigs_url":"https://www.ncbi.nlm.nih.gov/Traces/wgs/LXMK01","wgs_project_accession":"LXMK01"}}],"total_count": 1}

分区 基础知识 的第 169 页

ncbi-BLAST
15:51
2024年6月8日

帮助：BLAST+ 可执行文件 — BLASTHelp 文档 (nih.gov)

windows安装手册：Standalone BLAST Setup for Windows PC - BLAST® Help - NCBI Bookshelf (nih.gov)

Linux安装分析：Standalone BLAST Setup for Unix - BLAST® Help - NCBI Bookshelf (nih.gov)

conda install biopython::blast

建库手册：Create a masked BLAST database - BLAST® Command Line Applications User Manual - NCBI
Bookshelf (nih.gov)

结果展示：使用自定义输出格式显示 BLAST 搜索结果 - BLAST® 命令行应用程序用户手册 - NCBI Bookshelf
(nih.gov)

一、windows环境配置（不推荐）

Further configuring the PC will help facilitate the execution of blast+ programs and
streamline the access of installed databases. This configuration is through information
stored in special user environment variables. For 2.10.0 release of the blast+ package,
three variables are needed:

•

•

•

A modified path environment variable to indicate the location of installed blast+
programs, with "C:\users\taota\Desktop\blast-2.10.0+\bin\;" prepended to its existing
value
A new BLASTDB environment variable as pointer to database location, with "C:\users
\taota\Desktop\blast-2.10.0+\db\" as its value
A new BLASTDB_LMDB_MAP_SIZE, with 1000000 as its value (needed to
optimize makeblastdb operation when creating new database files)

来自 <https://www.ncbi.nlm.nih.gov/books/NBK52637/>

分区 基础知识 的第 170 页

二、建库
调用 makeblastdb 将新创建的 FASTA 序列文件转换为可爆破数据库 （M）。

makeblastdb -in sp_hs_subset.faa -dbtype prot -parse_seqids -title “demo: swissprot hs subset

without taxid” -out sp_hs_subset

此命令指示系统：

•

•

•

•

•

•

执行 makeblastdb 程序

使用 sp_hs_subset.faa 作为输入

将数据库类型设置为 protein

索引序列 ID（用于按 id 检索特定序列）

使用引号中的文本创建标题

重命名输出（否则将使用完整的 FASTA 文件名，包括文件扩展名）

使用 blastdbcmd（N，未显示）检查结果数据库。

来自 <https://www.ncbi.nlm.nih.gov/books/NBK52637/>

建库遇到如下问题

1.

2.

3.

以管理员身份打开cmd，运行makeblastdb

FASTA标准格式，一行不要太长
BLAST Database creation error: Near line 1, the local id is too long.  Its length is 76 but the maximum allowed local
id length is 50.  Please find and correct all local ids that are too long.
要添加 -parse_seqids

三、分析
由于RGDB已建好库，分别对应三个子库，因此默认evalue
blastp -db "RGDB-Mechanism_database\cardresis" -query "PLM-ARG_DB.fasta" -out PLM_RGDB_results.out

供参考：

四、Linux安装及使用示例
1.

blast --从建库到分析 | Hc's Blog (hc1023.github.io)

2.

Windows系统下本地Blast＋安装与使用图文教程 - 腾讯云开发者社区-腾讯云 (tencent.com)

分区 基础知识 的第 171 页

2.

Windows系统下本地Blast＋安装与使用图文教程 - 腾讯云开发者社区-腾讯云 (tencent.com)

1.
2.

每次打开前需export 指定程序及数据库位置

blastp -db "ARG-Database/RGDB-Mechanism_database/mibigresis" -query "ARG-
Database/PLM-ARG_DB.fasta" -out "ARG-Database/comparision/PLM-mibigresis.out"

结果解读

BLAST 输出自定义格式
-outfmt <String>
   alignment view options:
     0 = Pairwise,
     1 = Query-anchored showing identities,
     2 = Query-anchored no identities,
     3 = Flat query-anchored showing identities,
     4 = Flat query-anchored no identities,
     5 = BLAST XML,
     6 = Tabular,
     7 = Tabular with comment lines,
     8 = Seqalign (Text ASN.1),
     9 = Seqalign (Binary ASN.1),
    10 = Comma-separated values,

分区 基础知识 的第 172 页

    10 = Comma-separated values,
    11 = BLAST archive (ASN.1),
    12 = Seqalign (JSON),
    13 = Multiple-file BLAST JSON,
    14 = Multiple-file BLAST XML2,
    15 = Single-file BLAST JSON,
    16 = Single-file BLAST XML2,
    18 = Organism Report

./blastn -db ecoli -outfmt "7 qacc sacc evalue
qstart qend sstart send"

来自 <https://www.ncbi.nlm.nih.gov/books/NBK569862/>

   Options 6, 7 and 10 can be additionally configured to produce
   a custom format specified by space delimited format specifiers,
   or by a token specified by the delim keyword.
    E.g.: "10 delim=@ qacc sacc score".
   The delim keyword must appear after the numeric output format
   specification.
   The supported format specifiers are:
    qseqid means Query Seq-id
       qgi means Query GI
      qacc means Query accession
   qaccver means Query accession.version
      qlen means Query sequence length
    sseqid means Subject Seq-id
sallseqid means All subject Seq-id(s), separated by a ';'
       sgi means Subject GI
    sallgi means All subject GIs
      sacc means Subject accession
   saccver means Subject accession.version
   sallacc means All subject accessions
      slen means Subject sequence length
    qstart means Start of alignment in query
      qend means End of alignment in query
    sstart means Start of alignment in subject
      send means End of alignment in subject
      qseq means Aligned part of query sequence
      sseq means Aligned part of subject sequence
    evalue means Expect value
  bitscore means Bit score
     score means Raw score
    length means Alignment length
    pident means Percentage of identical matches
    nident means Number of identical matches
  mismatch means Number of mismatches
  positive means Number of positive-scoring matches
   gapopen means Number of gap openings
      gaps means Total number of gaps
      ppos means Percentage of positive-scoring matches
    frames means Query and subject frames separated by a '/'
    qframe means Query frame
    sframe means Subject frame
      btop means Blast traceback operations (BTOP)
    staxid means Subject Taxonomy ID
  ssciname means Subject Scientific Name
  scomname means Subject Common Name
sblastname means Subject Blast Name
sskingdom means Subject Super Kingdom
   staxids means unique Subject Taxonomy ID(s), separated by a ';'

(in numerical order)

sscinames means unique Subject Scientific Name(s), separated by a ';'
scomnames means unique Subject Common Name(s), separated by a ';'
sblastnames means unique Subject Blast Name(s), separated by a ';'

(in alphabetical order)

sskingdoms means unique Subject Super Kingdom(s), separated by a ';'

(in alphabetical order)

    stitle means Subject Title
salltitles means All Subject Title(s), separated by a '<>'
   sstrand means Subject Strand
     qcovs means Query Coverage Per Subject
   qcovhsp means Query Coverage Per HSP
    qcovus means Query Coverage Per Unique Subject (blastn only)

   When not provided, the default value is:
   'qaccver saccver pident length mismatch gapopen qstart qend sstart send
   evalue bitscore', which is equivalent to the keyword 'std'
   Default = `0'

分区 基础知识 的第 173 页

Prodigal
2024年5月14日

17:24

微课堂系列：Prodigal——国家微生物科学数据中心云工具
https://baijiahao.baidu.com/s?id=1694544946852112726
&wfr=spider&for=pc
1.原核生物mRNA特点

2.prodigal

分区 基础知识 的第 174 页

3.prodigal的安装

https://github.com/hyattpd/Prodigal/wiki/installation

4.输入文件

分区 基础知识 的第 175 页

4.输入文件

5.基本命令

分区 基础知识 的第 176 页

分区 基础知识 的第 177 页

6.输出结果

分区 基础知识 的第 178 页

核酸序列，星号表示终止子

7.在线平台

分区 基础知识 的第 179 页

首页-分析工具

文件系统

分区 基础知识 的第 180 页

好用的python包

2024年6月30日

13:35