分区 刘畅乐 的第 234 页

Implementation. Antibiotics, 2023. 12(3): p. 523.
2.    Cai, L., et al., Antimicrobial resistance bacteria and genes detected in hospital sewage provide valuable information in predicting clinical
antimicrobial resistance. Science of The Total Environment, 2021. 795: p. 148815.
3.    Oniciuc, E.A., et al., The Present and Future of Whole Genome Sequencing (WGS) and Whole Metagenome Sequencing (WMS) for Surveillance of
Antimicrobial Resistant Microorganisms and Antimicrobial Resistance Genes across the Food Chain. Genes, 2018. 9(5): p. 268.
4.    Ren, Y., et al., Prediction of antimicrobial resistance based on whole-genome sequencing and machine learning. Bioinformatics, 2021. 38(2): p.
325-334.
5.    Lakin, S.M., et al., Hierarchical Hidden Markov models enable accurate and diverse detection of antimicrobial resistance sequences.Communications
Biology, 2019. 2(1): p. 294.
6.    Davis, J.J., et al., Antimicrobial Resistance Prediction in PATRIC and RAST. Scientific Reports, 2016. 6(1): p. 27930.
7.    Arango-Argoty, G., et al., DeepARG: a deep learning approach for predicting antibiotic resistance genes from metagenomic data. Microbiome, 2018.
6(1): p. 23.
8.    Deng, F., et al., A comprehensive analysis of antibiotic resistance genes in the giant panda gut. iMeta, 2024. 3(1): p. e171.
9.    Ren, Y., et al., Deep Transfer Learning Enables Robust Prediction of Antimicrobial Resistance for Novel Antibiotics.Antibiotics, 2022. 11(11): p.
1611.
10.    Tang, L., Large models for genomics. Nature Methods, 2023. 20(12): p. 1868-1868.
11.    Yoo, H., et al. Predicting Anti-microbial Resistance using Large Language Models. 2023. arXiv:2401.00642 DOI: 10.48550/arXiv.2401.00642.
12.    Argimón, S., et al., Integrating whole-genome sequencing within the National Antimicrobial Resistance Surveillance Program in the Philippines.
Nature Communications, 2020. 11(1): p. 2719.
13.    Bennett, P.M., Plasmid encoded antibiotic resistance: acquisition and transfer of antibiotic resistance genes in bacteria.British Journal of
Pharmacology, 2008. 153(S1): p. S347-S357.
14.    Bhattacharyya, R.P., et al., Simultaneous detection of genotype and phenotype enables rapid and accurate antibiotic susceptibility determination.
Nature Medicine, 2019. 25(12): p. 1858-1864.
15.    Danko, D., et al., A global metagenomic map of urban microbiomes and antimicrobial resistance. Cell, 2021. 184(13): p. 3376-3393.e17.
16.    Johnson, K. and S. Lin, Call to work together on microarray data analysis. Nature, 2001. 411(6840): p. 885-885.
17.    Mangul, S., et al., Systematic benchmarking of omics computational tools. Nature Communications, 2019. 10(1): p. 1393.
18.    Going for algorithm gold. Nature Methods, 2008. 5(8): p. 659-659.

2.

研究内容

⚫

⚫

⚫

⚫

⚫

⚫
3.

⚫
4.

基因组/宏基因组组装与质量控制。

基因与抗性基因ARG的识别。与现有数据库（文献库）的对比分析。

基因组数据、元数据、SNP等多模态数据表征融合。

深度学习模型搭建，实现AMR表型预测。

利用模型结果，考虑AMR风险定义及预测机制。

AMR数据库与平台搭建。
拟解决的关键问题

同上。ARG基因组识别有没有什么特殊的地方？
研究方法

⚫

⚫

⚫
⚫

文献调研

横向对比

统计分析
深度学习

5.

研究特色及创新点

⚫

⚫

从测序数据、AMR预测到临床服务的全链路实践

关注计算机科学的最新进展，更新已有的AMR预测技术手段
在不完全了解ARG与耐药机制的背景下，可扩展的解决AMR预测问题

分区 刘畅乐 的第 235 页

中期答辩书撰写

2024年10月25日

20:18